third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/ConvertWarpSpecializeToLLVM.cpp

```diff
@@ -117,11 +118,14 @@ static LogicalResult rewriteWarpGroupBarriers(LLVM::LLVMFuncOp func,
     if (auto bar = dyn_cast<NVVM::Barrier0Op>(op)) {
       TritonLLVMIRRewriter b(bar.getLoc(), bar);
       createBarrier(b, /*barIdx=*/0, defaultWarpGroupSize, /*aligned=*/true);
-      bar.erase();
+      to_erase.push_back(bar);
```

Comment:
nit: it's actually better to return `WalkResult::skip()` instead

Reply:
then you can erase while walking preorder

Reply:
even better :)